### PR TITLE
CH suggestions

### PR DESCRIPTION
**Pull request by mnot**
Suggested elaboration / re-focusing in the introduction to clarify how CH fits into things.

### CODE REVIEW

**Comment by mnot** [comment]
CI failed due to rfc-editor.org DNS failure; should be transient.

File: draft-ietf-httpbis-client-hints.md [hunk 1]
```diff
@@ -54,9 +54,11 @@ informative:
 
 --- abstract
 
-An increasing diversity of Web-connected devices and software capabilities has created a need to deliver optimized content for each device.
+HTTP defines proactive content negotiation to allow servers to select the appropriate response for a given request, based upon the user agent's characteristics, as expressed in request headers. In practice, clients are often unwilling to send those request headers, because it is not clear whether they will be used, and sending them impacts both performance and privacy.
 
-This specification defines an extensible and configurable set of HTTP request header fields, colloquially known as Client Hints, to address this. They are intended to be used as input to proactive content negotiation; just as the Accept header field allows user agents to indicate what formats they prefer, Client Hints allow user agents to indicate device and agent specific preferences.
+This specification defines a response header, Accept-CH, that servers can use to advertise their use of request headers for proactive content negotiation, along with a set of guidelines for the creation of such headers, colloquially known as "Client Hints."
```
Comment (yoavweiss) [inline draft-ietf-httpbis-client-hints.md]: It also defines `Accept-CH-Lifetime`

File: draft-ietf-httpbis-client-hints.md [hunk 2]
```diff
@@ -81,11 +83,19 @@ One way to infer some of these capabilities is through User-Agent (Section 5.5.3
...
+A popular alternative strategy is to use HTTP cookies ({{RFC6265}}) to communicate some information about the user agent. However, this approach is also not cache friendly, bound by same origin policy, and often imposes additional client-side latency by requiring JavaScript execution to create and manage HTTP cookies.
+
+Proactive content negotiation (Section 3.4.1 of {{RFC7231}}) offers an alternative approach; user agents use specified, well-defined request headers to advertise their capabilities and characteristics, so that servers can select (or formulate) an appropriate response.
+
+However, proactive content negotiation requires clients to send these request headers prolifically. This causes performance concerns (because it creates "bloat" in requests), as well as privacy issues (because these request headers can be used to "fingerprint" -- i.e., uniquely identify -- the user agent).
```
Comment (yoavweiss) [inline draft-ietf-httpbis-client-hints.md]: s/used to "fingerprint"/used to passively "fingerprint"/?

File: draft-ietf-httpbis-client-hints.md [hunk 3]
```diff
@@ -81,11 +83,19 @@ One way to infer some of these capabilities is through User-Agent (Section 5.5.3
...
+
+This document defines a new response header, Accept-CH, that allows an origin server to explicitly ask that clients send these headers in requests, for a period of time bounded by the Accept-CH-Lifetime response header. It also defines guidelines for content negotiation mechanisms that use it, colloquially referred to as Client Hints.
```
Comment (yoavweiss) [inline draft-ietf-httpbis-client-hints.md]: Nit: should header names somewhat be marked as "code" or italic?

File: draft-ietf-httpbis-client-hints.md (hunk 1 above)
Comment (igrigorik) [reply draft-ietf-httpbis-client-hints.md]: +1

> ... defines a mechanism, based on `Accept-CH` and `Accept-CH-Lifetime` response headers, that servers ...

File: draft-ietf-httpbis-client-hints.md [hunk 4]
```diff
@@ -54,9 +54,11 @@ informative:
...
+HTTP defines proactive content negotiation to allow servers to select the appropriate response for a given request, based upon the user agent's characteristics, as expressed in request headers. In practice, clients are often unwilling to send those request headers, because it is not clear whether they will be used, and sending them impacts both performance and privacy.
 
-This specification defines an extensible and configurable set of HTTP request header fields, colloquially known as Client Hints, to address this. They are intended to be used as input to proactive content negotiation; just as the Accept header field allows user agents to indicate what formats they prefer, Client Hints allow user agents to indicate device and agent specific preferences.
+This specification defines a response header, Accept-CH, that servers can use to advertise their use of request headers for proactive content negotiation, along with a set of guidelines for the creation of such headers, colloquially known as "Client Hints."
+
+It also defines an initial set of Client Hints.
```
Comment (igrigorik) [inline draft-ietf-httpbis-client-hints.md]: I think our current plan is to omit it, actually.

File: draft-ietf-httpbis-client-hints.md (hunk 2 above)
Comment (igrigorik) [reply draft-ietf-httpbis-client-hints.md]: Perhaps...

> ... This causes performance concerns due to "bloat" of request headers, as well as privacy concerns due to passive fingerprinting that the data carried in these request headers may enable.

File: draft-ietf-httpbis-client-hints.md [hunk 5]
```diff
@@ -81,11 +83,19 @@ One way to infer some of these capabilities is through User-Agent (Section 5.5.3
...
+
+Proactive content negotiation (Section 3.4.1 of {{RFC7231}}) offers an alternative approach; user agents use specified, well-defined request headers to advertise their capabilities and characteristics, so that servers can select (or formulate) an appropriate response.
+
+However, proactive content negotiation requires clients to send these request headers prolifically. This causes performance concerns (because it creates "bloat" in requests), as well as privacy issues (because these request headers can be used to "fingerprint" -- i.e., uniquely identify -- the user agent).
+
+This document defines a new response header, Accept-CH, that allows an origin server to explicitly ask that clients send these headers in requests, for a period of time bounded by the Accept-CH-Lifetime response header. It also defines guidelines for content negotiation mechanisms that use it, colloquially referred to as Client Hints.
+
+Client Hints mitigate the performance concerns by assuring that clients will only send the request headers when they're actually going to be used, and the privacy concerns by creating positive evicence of their use (the use of the response header).
```
[the remark this answers is not in the document]
Comment (igrigorik) [reply draft-ietf-httpbis-client-hints.md]: > ... and privacy concerns of passive fingerprinting by requiring explicit opt-in and disclosure of required headers by the server through the use of the Accept-CH response header.

Maybe a bit more explicit and clearer?

File: draft-ietf-httpbis-client-hints.md (hunk 4 above)
Comment (reschke) [reply draft-ietf-httpbis-client-hints.md]: "header" -> "header field" (throughout)

File: draft-ietf-httpbis-client-hints.md (hunk 3 above)
Comment (mnot) [reply draft-ietf-httpbis-client-hints.md]: No; we don't have that luxury in RFCs.

File: draft-ietf-httpbis-client-hints.md (hunk 4 above)
Comment (mnot) [reply draft-ietf-httpbis-client-hints.md]: Let's separate those out.

**Comment by mnot** [comment]
Feedback addressed, I think.

**Pull request merged**